cratedb_toolkit/io/mongodb/extract.py

```diff
@@ -182,5 +182,17 @@ def extract_schema_from_array(array: list, schema: dict):
 }
 
 
-def get_type(o):
-    return TYPES_MAP.get(type(o), "UNKNOWN")
+def get_type(value):
+    """
+    Resolve value type via type map, with special treatment for integer types.
+
+    INTEGER: -2^31 to 2^31-1
+    BIGINT: -2^63 to 2^63-1
+    """
+    type_ = type(value)
```

Comment:
`isinstance(value, int)` isn't this better?

Reply:
`type_` is used to look up the target type from the defined map in `TYPES_MAP`.

https://github.com/crate/cratedb-toolkit/blob/d1c0740a218bba94dfe70d0e8786b4fdc5d4f1f0/cratedb_toolkit/io/mongodb/extract.py#L168-L182

Reply:
right